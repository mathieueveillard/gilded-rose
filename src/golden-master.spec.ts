import runGoldenMaster from 'jest-golden-master'
import { Item, GildedRose } from '.'

test("Ceci est un test", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Aged Brie", 10, 30);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})

test("sellIn inférieur à 0", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Aged Brie", -1, 60);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})

test("sellIn inférieur à 0 et nom != Brie/TAFKAL80ETC/Sulfuras", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Different", -1, 0);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})
test("sellIn inférieur à 0, nom =! sulfuras et quality > 0", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Different", -1, 60);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})

test("sellIn inférieur à 0, nom = TAFKAL80ETC", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Backstage passes to a TAFKAL80ETC concert", -1, 60);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})

test("sellIn inférieur à 0, nom != TAFKAL80ETC et quality <50", async () => {
    runGoldenMaster(async () => {
        const item1 = new Item("Backstage passes to a TAFKAL80ETC concert", -1, 40);
        const itemsArray = [item1];
        const gr = new GildedRose(itemsArray)
        gr.updateQuality()
    })
})