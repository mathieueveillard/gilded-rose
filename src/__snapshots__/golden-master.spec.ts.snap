// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ceci est un test 1`] = `
[
  "31 Quality is less than 50', 'Quality win 1",
  "Aged Brie is Not Sulfuras item so sell In decrease of 1 ",
]
`;

exports[`sellIn inférieur à 0 1`] = `
[
  "Aged Brie is Not Sulfuras item so sell In decrease of 1 ",
  "sell In inferior than 0",
]
`;

exports[`sellIn inférieur à 0 et nom != Brie/TAFKAL80ETC/Sulfuras 1`] = `
[
  "Different is not Aged Brie and not TAFKAL80ETC concert",
  "Different is Not Sulfuras item so sell In decrease of 1 ",
  "sell In inferior than 0",
  "Is not Brie",
  "Is not also TAFKAL80ETC concert",
]
`;

exports[`sellIn inférieur à 0, nom != TAFKAL80ETC et quality <50 1`] = `
[
  "41 Quality is less than 50', 'Quality win 1",
  "Backstage passes to a TAFKAL80ETC concert Is TAFKAL80ETC concert part",
  "sell In under 11",
  "42 Quality is less than 50 Quality win 1 from 11 sell In",
  "sell In under 6",
  "43 Quality is less than 50 Quality win 1 from 6 sell in",
  "Backstage passes to a TAFKAL80ETC concert is Not Sulfuras item so sell In decrease of 1 ",
  "sell In inferior than 0",
  "Is not Brie",
  "else quality - quality",
]
`;

exports[`sellIn inférieur à 0, nom = TAFKAL80ETC 1`] = `
[
  "Backstage passes to a TAFKAL80ETC concert is Not Sulfuras item so sell In decrease of 1 ",
  "sell In inferior than 0",
  "Is not Brie",
  "else quality - quality",
]
`;

exports[`sellIn inférieur à 0, nom =! sulfuras et quality > 0 1`] = `
[
  "Different is not Aged Brie and not TAFKAL80ETC concert",
  "60 Quality is more than 0",
  "Different Is not Sulfuras Weapon  59 quality has decrease of 1",
  "Different is Not Sulfuras item so sell In decrease of 1 ",
  "sell In inferior than 0",
  "Is not Brie",
  "Is not also TAFKAL80ETC concert",
  "quality is more than 0",
  "Is also not Sulfuras weapon",
  "the quality decrease of 1 because is something else",
]
`;
